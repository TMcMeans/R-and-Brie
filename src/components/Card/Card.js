import React from 'react';
import { Link } from 'react-router-dom';

export const Card = props => {
  const { image, label } = props;

  const cleanLabel = label.replace('recipes', '');
  const backgroundImage = {
    backgroundImage: `url(${image})`
  };

  return (
    <Link to={`/recipes/${label}`}>
      <div className="card" style={backgroundImage}>
        {/* <div className="fav-icon">
          <i className="fas fa-heart" />
        </div> */}
        <h3 className="label">{cleanLabel}</h3>
      </div>
    </Link>
  );
};
