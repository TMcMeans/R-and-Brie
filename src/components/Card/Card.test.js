import React from 'react'
import { shallow } from 'enzyme'
import { Card } from './Card'

describe('Card', () => {
  it('should render snapshot with all data passed in', () => {
    const mockRecipe = {
      image: "https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg",
      label: "Grilled Radicchio and Fontina"
    }
    const wrapper = shallow(<Card recipe={mockRecipe} />)
    expect(wrapper).toMatchSnapshot()
  })
})