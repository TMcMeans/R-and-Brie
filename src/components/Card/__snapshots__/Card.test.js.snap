// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    recipe={
      Object {
        "image": "https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg",
        "label": "Grilled Radicchio and Fontina",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="card"
        style={
          Object {
            "backgroundImage": "url(https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg)",
          }
        }
      >
        <h3
          className="label"
        >
          Grilled Radicchio and Fontina
        </h3>
      </div>,
      "replace": false,
      "to": "/recipes/Grilled Radicchio and Fontina",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="label"
        >
          Grilled Radicchio and Fontina
        </h3>,
        "className": "card",
        "style": Object {
          "backgroundImage": "url(https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Grilled Radicchio and Fontina",
          "className": "label",
        },
        "ref": null,
        "rendered": "Grilled Radicchio and Fontina",
        "type": "h3",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="card"
          style={
            Object {
              "backgroundImage": "url(https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg)",
            }
          }
        >
          <h3
            className="label"
          >
            Grilled Radicchio and Fontina
          </h3>
        </div>,
        "replace": false,
        "to": "/recipes/Grilled Radicchio and Fontina",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="label"
          >
            Grilled Radicchio and Fontina
          </h3>,
          "className": "card",
          "style": Object {
            "backgroundImage": "url(https://www.edamam.com/web-img/f37/f37bd3a564906b0fd489efc3bcde1dad.jpg)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Grilled Radicchio and Fontina",
            "className": "label",
          },
          "ref": null,
          "rendered": "Grilled Radicchio and Fontina",
          "type": "h3",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
